File: java/src/foo/Foo.java
package foo;

import java.util.Collections;
import java.util.List;

public final class Foo {

    public final int intValue;
    public final String strValue;
    public final List<Integer> listValue; 

    public Foo(final int intValue, final String strValue, final List<Integer> listValue) {
        this.intValue = intValue;
        this.strValue = strValue;
        this.listValue = Collections.unmodifiableList(listValue);
    }

    @Override
    public String toString() {
        final StringBuilder sb = new StringBuilder();
        sb.append("INT: ").append(intValue).append("\n")
          .append("STRING: ").append(strValue).append("\n")
          .append("LIST: ").append(listValue.toString());
        return sb.toString();
    }
}

File: golang/src/foo/foo.go
package foo

import "fmt"


type Foo struct {
	intValue int
	strValue string
	listValue []int
}

func (foo *Foo) IntValue() int {
	return foo.intValue;
}

func (foo *Foo) StrValue() string {
	return foo.strValue;
}

func (foo *Foo) ListValue() []int {
	return foo.listValue;
}

func (foo *Foo) String() string {
	result := fmt.Sprintf("INT: %d\nSTRING: %s\nLIST: [", foo.intValue, foo.strValue)
	for i, num := range foo.listValue {
		if i > 0 {
			result += ", "
		}
		result += fmt.Sprintf("%d", num)
	}
	result += "]"
	return result
}

func New(i int, s string, l []int) Foo {
	return Foo{intValue: i, strValue: s, listValue: l}
}
